ssl/statem/statem_lib.c

```diff
@@ -101,8 +101,8 @@ int tls_setup_handshake(SSL *s)
     memset(s->ext.extflags, 0, sizeof(s->ext.extflags));
 
     if (ssl_get_min_max_version(s, &ver_min, &ver_max, NULL) != 0) {
-        SSLfatal(s, SSL_AD_INTERNAL_ERROR, SSL_F_TLS_SETUP_HANDSHAKE,
-                    ERR_R_INTERNAL_ERROR);
+        SSLfatal(s, SSL_AD_PROTOCOL_VERSION, SSL_F_TLS_SETUP_HANDSHAKE,
```

Comment:
Does this turn this code path into an information leak?
That's the only reason I can think of for changing to internal error.

@mattcaswell ?

Reply:
> Does this turn this code path into an information leak?

I don't see one. The proposed change here looks reasonable.